wasm/jit/jit_arm64.go

```diff
@@ -327,16 +356,87 @@ func (c *arm64Compiler) compileCallIndirect(o *wazeroir.OperationCallIndirect) e
 	return fmt.Errorf("TODO: unsupported on arm64")
 }
 
+// compileDrop implements compiler.compileDrop for the arm64 architecture.
 func (c *arm64Compiler) compileDrop(o *wazeroir.OperationDrop) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	r := o.Range
+	if r == nil {
+		return nil
+	} else if r.Start == 0 {
+		// If the drop starts from the top, we just mark all the regiters
+		// used by drop targets unused.
+		for i := 0; i <= r.End; i++ {
+			if loc := c.locationStack.pop(); loc.onRegister() {
+				c.markRegisterUnused(loc.register)
+			}
+		}
+		return nil
+	}
+
+	// Save the non-drop target values.
+	liveValues := make([]*valueLocation, r.Start)
+	for i := 0; i < r.Start; i++ {
+		live := c.locationStack.pop()
+		liveValues[i] = live
+	}
+
+	// Then mark all registers used by drop tragets unused.
+	for i := 0; i < r.End-r.Start+1; i++ {
+		if loc := c.locationStack.pop(); loc.onRegister() {
+			c.markRegisterUnused(loc.register)
+		}
+	}
+
+	for i := range liveValues {
+		live := liveValues[len(liveValues)-1-i]
+		// If the value is on a memory, we have to move it to a register,
+		// otherwise the memory location is overriden by other values
+		// after this drop instructin.
+		if err := c.ensureOnGeneralPurposeRegister(live); err != nil {
+			return err
+		}
+		// Modify the stack location by pushing onto the location stack.
+		c.locationStack.push(live)
+	}
+	return nil
 }
 
 func (c *arm64Compiler) compileSelect() error {
 	return fmt.Errorf("TODO: unsupported on arm64")
 }
 
+// compilePick implements compiler.compilePick for the arm64 architecture.
 func (c *arm64Compiler) compilePick(o *wazeroir.OperationPick) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	c.maybeMoveTopConditionalToFreeGeneralPurposeRegister()
+
+	pickTarget := c.locationStack.stack[c.locationStack.sp-1-uint64(o.Depth)]
+	pickedRegister, err := c.allocateRegister(pickTarget.registerType())
+	if err != nil {
+		return err
+	}
+
+	if pickTarget.onRegister() {
+		// If the pick target is on a register, we simply copy the value
```

Comment:
since "if " already says pickTarget.onRegister. sometimes I use a trialing comment to save lines and doc drift.

ex. 

```
if pickTarget.onRegister() { // copy the value to the picked register
```